content/api/snippets.apib

```diff
@@ -0,0 +1,211 @@
+
+
++ Request (application/json)
+
+    + Body
+
+        ```
+        {
+            "id" : "header",
+            "name": "Header snippet",
+            "content": {
+                "html": "<b>Our cool header snippet for {{name}}</b>"
+            }
+        }
+        ```
+
++ Response 200 (application/json)
+
+        {
+          "results": {
+            "id": "header"
+          }
+        }
+
++ Response 422 (application/json)
+
+        {
+          "errors" : [
+            {
+              "part" : "text",
+              "message" : "Error while compiling text part: line 4: syntax error near 'age'",
+              "line" : 4
+            }
+          ]
+        }
+
+## Retrieve a Snippet [GET /snippets/{id}]
+
++ Parameters
+    + id (required, string, `ourfooter`)
+
++ Request
+
+
++ Response 200 (application/json)
+
+        {
+          "results" : {
+            "id" : "ourfooter",
+            "name": "Footer",
+            "content": {
+              "html" : "<b>Our standard footer</b>",
+              "text" : "Our standard footer",
+              "amp_html" : "<b>Our standard footer</b>"
+            },
+            "created_at": "2018-10-11T19:13:29.548Z",
+            "updated_at": "2018-10-11T19:14:50.181Z"
+          }
+        }
+
+## Update a Snippet [PUT /snippets/{id}]
+
+Unlike the [Templates API](api/templates/), if a content object is
+provided in the update request, it only needs to include the fields that
+are being updated. Fields that are not included will keep their current
+values.
+
+To remove a content field that was specified previously, simply set that
+field to an empty string in the update request.
+
++ Data Structure
+    + name (string) - Display name
+    + content (object) - HTML, AMPHTML, and/or text content. Substitution syntax is supported with the exception of the render_dynamic_content() and render_snippet() macro calls. In other words, snippets may not reference other snippets.
+        + html (string) - HTML content which will be used when the snippet is referenced from the HTML part of a template or inline content. Maximum length - 100KB.
+        + text (string) - Plain text content which will be used when the snippet is referenced from the plain text part or headers of a template or inline content. Maximum length - 100KB.
+        + amp_html (string) - AMPHTML content which will be used when the snippet is referenced from the AMPHTML part of a template or inline content. Maximum length - 100KB.
+    + shared_with_subaccounts (boolean) - Whether this snippet can be used by subaccounts.
+
+
++ Parameters
+    + id (required, string, `ourfooter`)
+
++ Request Update (application/json)
+
+    + Body
+
+        ```
+       {
+         "content": {
+           "html" : "<b>Our updated footer</b>",
+           "text" : "Our updated footer",
+           "amp_html" : "<b>Our updated footer</b>"
+          }
+        }
+        ```
+
++ Request Update to Remove a Content Part (application/json)
```

Comment:
I would be explicit on what part of the content you are removing here.  Makes sense to us (especially if you read the notes above), but if you just glance at it comparing to your example above, one might jump to the conclusion that you are removing `text`, not `html`.

Bring it up because I had to double look at it myself.

Reply:
Agreed, I think this can be clearer.